Effect.ts

```diff
@@ -24,36 +24,34 @@ export class Effect<
   ) {
     this.id = `${this.kind}(${this.children?.map(U.id.of).join(",") || ""})`;
   }
+
+  access = <K extends $<keyof T>>(
+    key: K,
+  ): Effect<T[U.AssertKeyof<T, K>], E, V> => {
+    return new Effect("Access", (process) => {
+      return U.memo(() => {
+        return U.thenOk(
+          U.all(process.resolve(this), process.resolve(key)),
+          ([target, key]) => target[key],
+        );
+      });
+    }, [this, key]);
+  };
+
+  as = <Satisfies extends T>() => {
```

Comment:
Why `Satisfies`?

Reply:
It's for narrowing, so the supplied type should ideally extend the pre-existing `T`, no? The word "satisfies" is also a gentle reminder for us to (possibly) eventually make use of TS 4.9's `satisfies` operator.

Reply:
`satisfies` is for safe upcasts, but this is an unsafe downcast. It should really be `U extends T` IMO.

Reply:
Fair enough, just updated